coordinator/src/node.rs

```diff
@@ -541,6 +646,233 @@ impl Node {
+                        // the accepting party has do bring in additional margin for the fees
+                        accept_collateral: margin_trader + fee,
+                        fee_rate,
+                        contract_infos: vec![ContractInputInfo {
+                            contract_descriptor,
+                            oracles: OracleInput {
+                                public_keys: vec![self.inner.oracle_pk()],
+                                event_id,
+                                threshold: 1,
+                            },
+                        }],
+                    }
+                };
+
+                tokio::spawn({
+                    let node = self.inner.clone();
+                    async move {
+                        if let Err(e) = node
+                            .propose_dlc_channel(channel_details.clone(), contract_input)
+                            .await
+                        {
+                            tracing::error!(
+                                channel_id = %channel_details.channel_id.to_hex(),
+                                peer_id = %node_id,
+                                "Failed to propose DLC channel as part of position resizing: {e:#}"
+                            );
+                            return;
+                        }
+
+                        let temporary_contract_id = match node
+                            .get_temporary_contract_id_by_sub_channel_id(channel_details.channel_id)
+                        {
+                            Ok(temporary_contract_id) => temporary_contract_id,
+                            Err(e) => {
+                                tracing::error!(
+                                    channel_id = %channel_details.channel_id.to_hex(),
+                                    "Unable to extract temporary contract id: {e:#}"
+                                );
+                                return;
+                            }
+                        };
+
+                        if let Err(e) = db::positions::Position::update_resized_position(
+                            &mut conn,
+                            old_position.trader.to_string(),
+                            total_contracts,
+                            direction.into(),
+                            leverage_coordinator,
+                            leverage_trader,
+                            margin_coordinator as i64,
+                            margin_trader as i64,
+                            f32_from_decimal(average_execution_price),
+                            f32_from_decimal(liquidation_price_trader),
+                            expiry_timestamp,
+                            temporary_contract_id,
+                            stable,
+                        ) {
+                            tracing::error!(
+                                channel_id = %channel_details.channel_id.to_hex(),
+
```

Comment:
nit: is this an extra new line or is it just badly shown on my phone 😅

Reply:
Good catch, didn't mean that.